src/main/java/com/sumologic/log4j/aggregation/SumoBufferFlusher.java

```diff
@@ -81,4 +101,12 @@ public void stop() throws InterruptedException
       logger.warn("Timed out waiting for buffer flusher to finish.");
     }
   }
+
+  public void flushOnError()
+  {
+    if(this.flushOnError && !this.errorLogObserver.hasError()) {
+      this.errorLogObserver.recordError(true);
+      flushingThread.interrupt();
```

Comment:
need an opinion on using interrupt here. I think it might cause a problem because SumoHttpSender will block while sending logs to Sumo server and interrupt will abort the transfer. we can simply set the error flag to true, and let the flusher flushes data during the next interval.